dulwich/repo.py

```diff
@@ -676,10 +677,17 @@ def __init__(self, root):
             raise NotGitRepository(
                 "No git repository was found at %(path)s" % dict(path=root)
             )
+        worktree_commondir = os.path.join(self._controldir, COMMONDIR)
+        if(os.path.isfile(worktree_commondir)):
+            with open(worktree_commondir, 'r') as f:
+                self._commondir = os.path.join(self.controldir(),\
```

Comment:
Please just use self._get_named_file(), which also works for e.g. subclasses that aren't using local disk.